File: exercise/java/day02/src/main/java/games/FizzBuzz.java
package games;

public class FizzBuzz {
    private FizzBuzz() {
    }
    
    public static String convert(Integer input) throws OutOfRangeException {
        if (isBetween(input, 0, 100) {
            if (input % 3 == 0 && input % 5 == 0) {
                return "FizzBuzz";
            }
            if (input % 3 == 0) {
                return "Fizz";
            }
            if (input % 5 == 0) {
                return "Buzz";
            }
            return input.toString();
        } else {
            throw new OutOfRangeException();
        }
    }

    private static boolean isBetween(Integer input, Integer min, Integer max) {
        return input >= min && input <= max;
    }
}

File: exercise/kotlin/day02/src/main/kotlin/games/FizzBuzz.kt
package games

object FizzBuzz {
    fun convert(input: Int): String {
        return if (input > 0) {
            if (input <= 100) {
                if (input % 3 == 0 && input % 5 == 0) {
                    return "FizzBuzz"
                }
                if (input % 3 == 0) {
                    return "Fizz"
                }
                if (input % 5 == 0) {
                    "Buzz"
                } else input.toString()
            } else {
                throw OutOfRangeException()
            }
        } else {
            throw OutOfRangeException()
        }
    }
}

